airflow/contrib/hooks/bigquery_hook.py

```diff
@@ -238,6 +238,8 @@ def create_empty_table(self,
 
         :return:
         """
+        if time_partitioning is None:
+            time_partitioning = dict()
```

Comment:
What do you mean? Setting the `{}` in the arguments is bad practice by the way Python creates these objects.

Reply:
@Fokko No no, it is a bad practise to use `{}` or `dict()` as a default argument. Reference: https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

The above change is because we had the below code:
![image](https://user-images.githubusercontent.com/8811558/43825996-141df72c-9aee-11e8-8b2f-c55f383f5ae9.png)

which I changed to the following:
![image](https://user-images.githubusercontent.com/8811558/43826016-1f5c03a4-9aee-11e8-8c2c-06e2f715ae33.png)

Reply:
`{}` would probably be better in hindsight, but there's not much in it. On my laptop:

```
In [5]: %timeit {}
47.2 ns ± 2.3 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [6]: %timeit dict()
168 ns ± 1.77 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```

so yes dict() is 3 times as "slow" as `{}`, but neither is particularly slow.

Reply:
@xnuinside Hi, there is no specific reason for me to use `{}` compared to `dict()`. The point for this change as I mentioned earlier was to remove an empty list from default arguments. 

And as Ash pointed out `{}` dict literal is faster that dict constructor `dict()` but there is not huge difference as the dict builds up. Check out the below links for more detailed read:
- https://doughellmann.com/blog/2012/11/12/the-performance-impact-of-using-dict-instead-of-in-cpython-2-7-2/
- https://stackoverflow.com/questions/6610606/is-there-a-difference-between-using-a-dict-literal-and-a-dict-constructor

Reply:
@kaxil , @ashb , thx!